### ISSUE TITLE
covpass-sdk-demo is not defaulting to test-environment

### ISSUE THREAD

**Issue by pc-coholic** [Digitaler-Impfnachweis/covpass-android#38]
While integrating the covpass-sdk into our own scanning app, [pretixSCAN](https://github.com/pretix/pretixSCAN-android), I realized that no matter what I was doing, I was always hitting the production servers at Ubirch.

To include the covpass-sdk into our app, our `build.gradle`-file has two implementation lines:
```
    debugImplementation 'com.ibm.health.de.rki.covpass:covpass-sdk-demo:1.38.0'
    releaseImplementation 'com.ibm.health.de.rki.covpass:covpass-sdk:1.38.0'
```

To make sure that the issue wasn't just with the debug/releaseImplementation, I also tried using the classic implementation line:
```
    implementation 'com.ibm.health.de.rki.covpass:covpass-sdk-demo:1.38.0'
```

But to no avail - we are still hitting the PROD-server.

Just to be 100% certain, that this isn't a problem with our app, I went ahead and created a super simple test application which basically exclusively consists of a boilerplate Android Studio app with the covpass-sdk included as mentioned above: https://github.com/pc-coholic/covpass-sdk-demo_test

If you check out this repo and build the app, you will see that despite requesting the covpass-sdk-demo, we are still getting the PROD-server, as can be seen by the `trustServiceHost` that I'm surfacing from the `sdkDeps`.

It is worth noting, that this is not an issue when building the covpass-apps directly from your monolithic repository - this seems to only affect the individual libraries published to (GH) maven.

**Comment by oliver-steinbrecher** [comment]
Feel free to suggest a source change via PR :)

**Comment by pc-coholic** [comment]
> Feel free to suggest a source change via PR :)

If I knew why this happens, I would have done it already :)

**Comment by oliver-steinbrecher** [comment]
@wkornewald can you please provide support here?

**Comment by wkornewald** [comment]
@pc-coholic Why do you even want to use the demo backend in the first place? Do you have access to that backend at all?

**Comment by pc-coholic** [comment]
It's mainly a testing issue - using the contents of https://github.com/eu-digital-green-certificates/dgc-testdata/ it is a lot easier to test all kinds of different scenarios.

And on the other hand, I think I saw a note somewhere in covpass saying dev and testing should not be done against prod in order to not flood the system with constant download requests for the dsc-list...

**Comment by wkornewald** [comment]
This will be fixed in one of the next releases. Our modules depended on the BOM and that causes the dependencies to become flat, so resource merging is done in some arbitrary order.

**Comment by jleufgen** [comment]
@pc-coholic can we close this ticket?

**Comment by jleufgen** [comment]
@pc-coholic can we close this ticket?

**Comment by pc-coholic** [comment]
Sure, go ahead :-)

**Comment by wkornewald** [comment]
The fix release will be 1.11.x.
